pageserver/src/object_repository.rs

```diff
@@ -272,7 +273,7 @@ impl Timeline for ObjectTimeline {
 
                     self.put_page_image(tag, lsn, page_img.clone())?;
                 }
-                x => bail!("Unexpected object value: {:?}", x),
+                _ => page_img = Bytes::from_static(&ZERO_PAGE),
```

Comment:
I suggest handling ObjectValue explicitly here. We still don't expect to meet TimelineMetadata here, do we?

And write a comment on why it is ok to return zero-page for each case. I suspect it is not always ok and may require some extra checks.